Stratis.Bitcoin/BlockPulling/DownloadAssignmentStrategy.cs

```diff
@@ -0,0 +1,93 @@
+﻿using NBitcoin.Protocol;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using System.Collections.Concurrent;
+using NBitcoin.Protocol.Behaviors;
+using System.Threading;
+using Stratis.Bitcoin.Connection;
+
+namespace Stratis.Bitcoin.BlockPulling
+{
+    /// <summary>
+    /// Implements a strategy for a block puller that needs to download 
+    /// a set of blocks and from its connected peers.
+    /// </summary>
+    public static class DownloadAssignmentStrategy
```

Comment:
OK, I'll change it to PullerDownloadAssignments. 

In my view it is a strategy by its nature, regardless of whether we support other strategies to be implemented or not (in long term, everything can be changed to support another strategy).